RecoCTPPS/PixelLocal/src/RPixPlaneCombinatoryTracking.cc

```diff
@@ -0,0 +1,483 @@
+        //This avoid to convert the global plane-line interseption in order not to call the the geometry 
+        TVectorD maxGlobalPointDistance(0,2,maximumXLocalDistanceFromTrack_,maximumYLocalDistanceFromTrack_,0.,"END");
+        TMatrixD tmpPlaneRotationMatrixMap = planeRotationMatrixMap_[tmpPlaneId];
+        maxGlobalPointDistance = tmpPlaneRotationMatrixMap * maxGlobalPointDistance;
+        //I avoid the Sqrt since it will not be saved
+        double_t maximumXdistance = maxGlobalPointDistance[0]*maxGlobalPointDistance[0];
+        double_t maximumYdistance = maxGlobalPointDistance[1]*maxGlobalPointDistance[1];
+        double_t minimumDistance = 1. + maximumXdistance + maximumYdistance; // to be sure that the first min distance is from a real point
+        for(const auto & hit : hitMap_[tmpPlaneId]){
+          double xResidual = hit.globalPoint.x() - pointOnDet.X();
+          double yResidual = hit.globalPoint.y() - pointOnDet.Y();
+          double xDistance = xResidual*xResidual;
+          double yDistance = yResidual*yResidual;
+          double distance = xDistance + yDistance;
+          if(xDistance < maximumXdistance && yDistance < maximumYdistance && distance < minimumDistance){
+            std::pair<double,double> residuals = std::make_pair(xResidual,yResidual);
+            TMatrixD globalError = hit.globalError;
+            std::pair<double,double> pulls = std::make_pair(xResidual/TMath::Sqrt(globalError[0][0]),yResidual/TMath::Sqrt(globalError[1][1]));
+            delete fittedRecHit;
+            fittedRecHit = new CTPPSPixelLocalTrack::CTPPSPixelFittedRecHit(hit.recHit, pointOnDet, residuals, pulls);
+            fittedRecHit->setIsRealHit(true);
+          }
+        }
+      }
+      else{
+        fittedRecHit->setIsRealHit(false);
+        fittedRecHit->setGlobalCoordinates(pointOnDet);
+      }
+
+      bestTrack.addHit(tmpPlaneId, *fittedRecHit);
+  
+    }
+
+    localTrackVector_.push_back(bestTrack);
+
+    int pointForTracking = 0;
+    int pointOnTrack = 0;
+
+    if(verbosity_>=1){
+      for(const auto & planeHits : bestTrack.getHits()){
+        for(const auto & fittedhit : planeHits){
+          if(fittedhit.getIsUsedForFit()) ++pointForTracking;
+          if(fittedhit.getIsRealHit()) ++pointOnTrack;
+        }
+      }
+      std::cout<<"Best track has "<<pointForTracking<<" points used for the fit and " << pointOnTrack<<" points belonging to the track\n";
+    }
+
+  } //close of the while loop on all the hits
+
+  return;
+
+}
+
+//------------------------------------------------------------------------------------------------//
+
+CTPPSPixelLocalTrack RPixPlaneCombinatoryTracking::fitTrack(std::vector<RPixDetPatternFinder::PointInPlane> pointList){
+  
+  uint32_t numberOfPoints = pointList.size();
+  TVectorD xyCoordinates(2*numberOfPoints);
+  TMatrixD varianceMatrix(2*numberOfPoints,2*numberOfPoints);
+  TMatrixD zMatrix(2*numberOfPoints,4);
```

Comment:
Aren't SMatrix-based classes only usable when the size is known at compile time?

Reply:
[a response to this got lost]
Since the set of possible numberOfPoints is short, it should be easy to use fixed size matrices with a template and a switch.
Something to consider in the future, perhaps